### PR TITLE
DE-530 - Delete removed links and related AutomationConditionals

### PR DESCRIPTION
**Pull request by andresmoschini**
Hi team!

This is the query to delete the links removed from the campaign content.

It was easier than insert because of a [Dapper feature](https://github.com/DapperLib/Dapper#list-support). It is not so predictable, so I have added a code comment with the explanation.

It is pending:
- [x] to remove the links from automation conditionals

### CODE REVIEW

File: Doppler.HtmlEditorApi/Storage.DapperProvider/Queries/DeleteRemovedCampaignLinks.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+///       AND [Link].UrlLink NOT IN (@Links1,@Links2,@Links3,@Links4)',
+///     N'@IdContent int,@Links1 nvarchar(4000),@Links2 nvarchar(4000),@Links3 nvarchar(4000),@Links4 nvarchar(4000)',
+///     @IdContent=36357513,@Links1=N'https://www.google.com?q=|*|320*|*',@Links2=N'a',@Links3=N'b',@Links4=N'c'
+/// Example without links:
+///     exec sp_executesql N'
+///       DELETE FROM [Link]
+///       WHERE [Link].IdCampaign = @IdContent
+///       AND [Link].UrlLink NOT IN (SELECT @Links WHERE 1 = 0)',
+///     N'@IdContent int,@Links nvarchar(4000)',
+///     @IdContent=36357513,@Links=NULL
```
Comment (andresmoschini) [inline Doppler.HtmlEditorApi/Storage.DapperProvider/Queries/DeleteRemovedCampaignLinks.cs:25]: @MarianoHerrera maybe you are interested on review these queries.

Comment (MarianoHerrera) [reply Doppler.HtmlEditorApi/Storage.DapperProvider/Queries/DeleteRemovedCampaignLinks.cs:25]: The example isn't acurate you have a list of links and not a "select" with "1=0" (never will return anithing)

**Review by MarianoHerrera — APPROVED** [review]
I'm asuming this delete will be executed before insertion to work.

File: Doppler.HtmlEditorApi/Storage.DapperProvider/Queries/DeleteRemovedCampaignLinks.cs (hunk 1 above)
Comment (andresmoschini) [reply Doppler.HtmlEditorApi/Storage.DapperProvider/Queries/DeleteRemovedCampaignLinks.cs:25]: Right, so, in this case, it should delete all the links with `IdCampaign = @IdContent`, so, it is fine.

File: Doppler.HtmlEditorApi/Storage.DapperProvider/DapperCampaignContentRepository.cs [hunk 2]
```diff
@@ -136,4 +136,15 @@ public async Task SaveNewFieldIds(int ContentId, IEnumerable<int> fieldsId)
...
+
+    public async Task SaveLinks(int ContentId, IEnumerable<string> links)
+    {
+        if (links.Any())
+        {
+            // TODO: insert new links
+        }
+        // TODO: Delete removed links from AutomationConditional table
+        // See https://github.com/MakingSense/Doppler/blob/ed24e901c990b7fb2eaeaed557c62c1adfa80215/Doppler.Application.CampaignsModule/Services/Classes/CampaignContentService.cs#L2519-L2523
+        await _dbContext.ExecuteAsync(new DeleteRemovedCampaignLinks(ContentId, links));
```
Comment (andresmoschini) [inline Doppler.HtmlEditorApi/Storage.DapperProvider/DapperCampaignContentRepository.cs:148]: @MarianoHerrera 

> I'm asuming this delete will be executed before insertion to work.

The delete is being done after insert, but I do not see a problem there. What am I missing?

Comment (MarianoHerrera) [reply Doppler.HtmlEditorApi/Storage.DapperProvider/DapperCampaignContentRepository.cs:148]: Sorry, my bad..... It was just to see if you were paying attention 
giphy 1

File: Doppler.HtmlEditorApi/Storage.DapperProvider/Queries/DeleteAutomationConditionalsOfRemovedCampaignLinks.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+) : IExecutableDbQuery
+{
+    public string GenerateSqlQuery() => @"
+DELETE ac
+FROM [AutomationConditional] ac
+INNER JOIN [Link] l ON ac.IdLink = l.IdLink
+WHERE l.IdCampaign = @IdContent
+AND l.UrlLink NOT IN @Links
+";
+}
```
Comment (andresmoschini) [inline Doppler.HtmlEditorApi/Storage.DapperProvider/Queries/DeleteAutomationConditionalsOfRemovedCampaignLinks.cs:43]: @leoslopez @MarianoHerrera this is the query over `AutomationConditional` I have talked about.

**Review by maurocardinali — COMMENTED** [review]
Great! 🚀

**Pull request merged**